### ISSUE TITLE
[Bug]: Submissions page does not work for certain senders

### ISSUE THREAD

**Issue by arnejduranovic** [CDCgov/prime-reportstream#10415]
## Describe the bug
When navigating to the Submissions page on the website **as an organization that does not have a sender named "default"** : The API call returns an error but the UI does not register it, it just keeps the loading icon spinning forever

## Impact on ReportStream
Some senders cannot view their submissions. Additionally, Prime-Admins may be confused because if they are a STLT they don't have any senders but they can still go to the submissions page and just watch it load forever. However, this does not impact our external users (receivers).

## Steps to reproduce
1. Go to Staging RS
3. Set organization as Flexion
4. Go to submissions page, see it times out

This also reproduces locally

## Expected behavior
There are two user flows here:

### The org does not have any senders, in other words, the org is a receiver
In this case, Submissions tab should be hidden from the UI. If you don't want to do that, then do something other than have the page load forever

### The org has one or more senders, but none of them are named "default"
The submissions endpoint takes an organization name as a path parameter. IF you pass in just the org name for this param, as the front end does, then the backend will look for a sender called "default". "default" is just the name that gets assigned to a sender upon creation if a name is not specified, so in some cases, like in the case of the Flexion org, the endpoint returns not found (but UI does not update to reflect this error, it just keeps the loading animation). So, my proposal is the front end do the following:

When a user navigates to the submissions page, allow them to choose from a list of senders they have configured, THEN call the endpoint with the following organization path param: `<organization>.<sender_name>`. The backend will then parse this string and instead of trying to find a sender with the name "default" it will instead look for a sender with sender name `<sender_name>`. 

So, in the case of Flexion org in staging:
1. submitting `https://staging.prime.cdc.gov/api/waters/org/flexion/submissions?pageSize=61&cursor=3000-01-01T00%3A00%3A00.000Z&since=2000-01-01T00%3A00%3A00.000Z&until=3000-01-01T00%3A00%3A00.000Z&sortcol=undefined&sortdir=DESC&showfailed=false` is **WRONG**
2. submitting `https://staging.prime.cdc.gov/api/waters/org/flexion.simulated-hospital/submissions?pageSize=61&cursor=3000-01-01T00%3A00%3A00.000Z&since=2000-01-01T00%3A00%3A00.000Z&until=3000-01-01T00%3A00%3A00.000Z&sortcol=undefined&sortdir=DESC&showfailed=false` is **CORRECT**

**Comment by chris-kuryak** [comment]
This appears related to #10387 -- we will be discussing in standup today

**Comment by penny-lischer** [comment]
@arnejduranovic Questions for you:

1. We currently have organizations with senders named "default", how are these working?
2. When setting up an organization, and specifying a sender, does the sender value need to be a valid sender name? (If so, we could get 404 errors from the endpoint in this scenario as well).
@KeithNava @chris-kuryak

**Comment by KeithNava** [comment]
Also, since this is a change to the contract between the frontend and backend, it would be better categorized as a new issue/feature rather than a bug. This would ensure we have the right folks involved in the discussion to refine the request. cc: @chris-kuryak

**Comment by chris-kuryak** [comment]
TBD from meeting on Monday

**Comment by arnejduranovic** [comment]
@chris-kuryak  There is a work around for this so ELIMS pilot is not blocked by this work

Work around: Just have at least one sender named "default"

**Comment by chris-kuryak** [comment]
@arnejduranovic Thanks for the update! We'll be discussing this week as a team regarding a longer term solution and get back to you. Thank you again for the feedback yesterday.

**Comment by chris-kuryak** [comment]
@arnejduranovic 
Two quick questions/clarifications:

1. Did you mean to write "Senders" here instead of "Receivers"? Receivers don't have submissions in the COVID pipeline:

> Some receivers cannot view their submissions.

2. Regarding your statement in the ticket that:

> Additionally, some users may be confused because if they are a STLT they don't have any senders but they can still go to the submissions page and just watch it load forever...

Were you logged in as a Prime-Admin when you did this? I don't think that is true when you are logged in as specifically a receiver (not a Prime-admin duping themselves as a receiver, which I agree you will see all menus).
When a Receiver logs in, they see a different menu from everyone else and does not include submissions. Here is what they see:

![image.png](https://images.zenhubusercontent.com/637d310dba428f3e3d55c9db/28f10b9a-b3e2-408d-867b-e1f217cc5e3c)

Let me know if you still see this issue?

**Comment by arnejduranovic** [comment]
@chris-kuryak Yes I meant senders and I was logged in as admin and duped as an org, so no worries there then.

**Comment by chris-kuryak** [comment]
Thanks!

**Comment by chris-kuryak** [comment]
@oslynn  Just checking in on this bug. 
Should Flexion's submissions page have data in it?

Arnej flagged this a long time ago and I want to make sure I still understand the issue.

**Comment by oslynn** [comment]
@chris-kuryak @penny-lischer Currently the Admin UI only allows us to set <organization> when we click [ Set ] on the Organizations page as shown below.  When it queries the submission, it sends only the <organization> to the backend.  When the backend sees only org, it assumes the send_name to be default since it expects <organization>.<sender_name> to be sent.  As explained in the ticket description.  
<img width="2000" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBL01ZQkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--fa10c6050b58d47c28e9d83255175ef1d9961206/image.png" alt="image.png" />

@arnejduranovic provided a workaround for a short-term solution for ELIMS.  For a long-term solution, I suggest making modifications to the Frontend UI to send <organization>.<sender_name> when query sender submission.

**Comment by oslynn** [comment]
@chris-kuryak For the short term, every sender must have one entry name: default as shown in epic setting below:
<img width="504" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBL3dZQkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--850d5ea675ad0c23ea71766791e055883eb3b587/image.png" alt="image.png" />

**Comment by oslynn** [comment]
After adding the sender name: default, the submission page will show as below:
<img width="2000" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBLzhZQkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--7a31d1e0dfafc9c07076d9c5da8a6a407426be29/image.png" alt="image.png" />

Note.  Flexion doesn't have a sender name: default entry, it will show blank even it has messages sent.

<img width="2000" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBd0laQkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--6175dcd5cfc68c4c30a94ffa4caacae502cac4f1/image.png" alt="image.png" />

**Comment by chris-kuryak** [comment]
This issue only impacts orgs with multiple senders.
Currently, there are only two orgs (that we could find) with multiple senders:
- SimpleReport
- Flexion

**Comment by oslynn** [comment]
Done investigated.  Wait for a decision.
